src/views/dialogs/SharePNLAnalyticsDialog.tsx

```diff
@@ -0,0 +1,306 @@
+import { useMemo } from 'react';
+
+import { useToBlob } from '@hugocxl/react-to-image';
+import styled from 'styled-components';
+
+import { AbacusPositionSides, Nullable } from '@/constants/abacus';
+import { ButtonAction } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+import { PositionSide } from '@/constants/trade';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { LogoIcon } from '@/icons/logo';
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AssetIcon } from '@/components/AssetIcon';
+import { Button } from '@/components/Button';
+import { Dialog } from '@/components/Dialog';
+import { Icon, IconName } from '@/components/Icon';
+import { Output, OutputType, ShowSign } from '@/components/Output';
+import { QrCode } from '@/components/QrCode';
+import { Tag, TagSign } from '@/components/Tag';
+
+import { useAppDispatch } from '@/state/appTypes';
+import { closeDialog } from '@/state/dialogs';
+
+import { MustBigNumber } from '@/lib/numbers';
+import { triggerTwitterIntent } from '@/lib/twitter';
+
+type ElementProps = {
+  marketId: string;
+  assetId: string;
+  leverage: Nullable<number>;
+  oraclePrice: Nullable<number>;
+  entryPrice: Nullable<number>;
+  unrealizedPnlPercent: Nullable<number>;
+  side: Nullable<AbacusPositionSides>;
+  sideLabel: Nullable<string>;
+  setIsOpen: (open: boolean) => void;
+};
+
+const copyBlobToClipboard = async (blob: Blob | null) => {
+  if (!blob) {
+    return;
+  }
+
+  const item = new ClipboardItem({ 'image/png': blob });
+  await navigator.clipboard.write([item]);
+};
+
+export const SharePNLAnalyticsDialog = ({
+  marketId,
+  assetId,
+  side,
+  sideLabel,
+  leverage,
+  oraclePrice,
+  entryPrice,
+  unrealizedPnlPercent,
+  setIsOpen,
+}: ElementProps) => {
+  const stringGetter = useStringGetter();
+  const dispatch = useAppDispatch();
+
+  const [{ isLoading: isCopying }, convert, ref] = useToBlob<HTMLDivElement>({
+    quality: 1.0,
+    onSuccess: copyBlobToClipboard,
+  });
+
+  const [{ isLoading: isSharing }, convertShare, refShare] = useToBlob<HTMLDivElement>({
+    quality: 1.0,
+    onSuccess: async (blob) => {
+      await copyBlobToClipboard(blob);
+
+      triggerTwitterIntent({
+        text: `Check out my ${assetId} position on @dYdX\n\n#dYdX #${assetId}\n[paste image and delete this!]`,
```

Comment:
[q] is `[paste image and delete this!]` supposed to be in here?

Reply:
Yes